analyzers/src/SonarAnalyzer.CSharp/Rules/AvoidLambdaExpressionInLoopsInBlazor.cs

```diff
@@ -0,0 +1,86 @@
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+namespace SonarAnalyzer.Rules.CSharp;
+
+[DiagnosticAnalyzer(LanguageNames.CSharp)]
+public sealed class AvoidLambdaExpressionInLoopsInBlazor : SonarDiagnosticAnalyzer
+{
+    private const string DiagnosticId = "S6802";
+    private const string MessageFormat = "Avoid using lambda expressions in loops in Blazor markup.";
+
+    private static readonly DiagnosticDescriptor Rule = DescriptorFactory.Create(DiagnosticId, MessageFormat);
+
+    private static readonly ISet<string> AddAttributeMethods = new HashSet<string> { "AddAttribute", "AddMultipleAttributes" };
+
+    public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+    protected override void Initialize(SonarAnalysisContext context) =>
+        context.RegisterCompilationStartAction(cc =>
+        {
+            // If we are not in a Blazor project, we don't need to register for lambda expressions.
+            if (cc.Compilation.GetTypeByMetadataName(KnownType.Microsoft_AspNetCore_Components_Rendering_RenderTreeBuilder) is null)
+            {
+                return;
+            }
+
+            cc.RegisterNodeAction(c =>
+                {
+                    var node = (LambdaExpressionSyntax)c.Node;
+
+                    if (IsWithinLoop(node)
+                        && IsWithinRenderTreeBuilderInvocation(node, c.SemanticModel))
+                    {
+                        c.ReportIssue(Diagnostic.Create(Rule, node.GetLocation()));
+                    }
+                },
+                SyntaxKind.SimpleLambdaExpression,
+                SyntaxKind.ParenthesizedLambdaExpression);
+        });
+
+    private static bool IsWithinRenderTreeBuilderInvocation(SyntaxNode node, SemanticModel semanticModel)
+    {
+        while (node != null)
+        {
+            if (node is InvocationExpressionSyntax invocation
+                && AddAttributeMethods.Contains(invocation.GetName())
+                && semanticModel.GetSymbolInfo(invocation.Expression).Symbol is IMethodSymbol symbol
+                && symbol.ContainingType.GetSymbolType().Is(KnownType.Microsoft_AspNetCore_Components_Rendering_RenderTreeBuilder))
+            {
+                return true;
+            }
+            node = node.Parent;
+        }
+        return false;
+    }
+
+    private static bool IsWithinLoop(SyntaxNode node)
+    {
+        while (node != null)
+        {
+            if (node is ForStatementSyntax or ForEachStatementSyntax or WhileStatementSyntax or DoStatementSyntax)
+            {
+                return true;
+            }
+            node = node.Parent;
+        }
+        return false;
+    }
```

Comment:
What about:
```suggestion
    private static bool IsWithinLoop(SyntaxNode node) =>
        node.AncestorsAndSelf().Any(x => x is ForStatementSyntax or ForEachStatementSyntax or WhileStatementSyntax or DoStatementSyntax);
```
this can also be inlined